military_features/scripts/CalcSIDCField.py

```diff
@@ -0,0 +1,226 @@
+#----------------------------------------------------------------------------------
+# Copyright 2013 Esri
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#----------------------------------------------------------------------------------
+# CalcSIDCField.py
+# Description: Sets the SIDC field from the Representation Rule name 
+# Requirements: ArcGIS Desktop Standard
+#----------------------------------------------------------------------------------
+
+import os
+import sys
+import traceback
+import arcpy
+import MilitaryUtilities
+import DictionaryConstants
+
+### Params:
+### 0 - inputFC
+### 1 - sidc field
+### 2 - echeclon field
+### 3 - affiliation ("NOT_SET", "FRIENDLY", "HOSTILE", "NEUTRAL", "UNKNOWN")
+
+try :
+
+    # Get input feature class
+    inputFC = arcpy.GetParameter(0)
+    if (inputFC is "" or inputFC is None):
+        # Assumes TestData folder in same directory as .py
+        inputFC = os.path.join(os.path.dirname(__file__),  "TestData/Test_Data_No_Sidc.gdb/FriendlyOperations/FriendlyOperationsL")
+
+    if not arcpy.Exists(inputFC) :
+        msg = "Input Dataset does not exist: " + str(inputFC) + " - exiting"
+        arcpy.AddError(msg)
+        raise IOError(msg) 
+    
+    desc = arcpy.Describe(inputFC)
+
+    # Get input feature class
+    SIDCField = arcpy.GetParameterAsText(1)
+    if (SIDCField == "" or SIDCField is None):
+        SIDCField = "sidc"
+
+    standard = arcpy.GetParameterAsText(2)
+    
+    symbolDictionary = MilitaryUtilities.getSymbolDictionaryStandard(standard)
+    
+    # Get input feature class
+    EchelonField = arcpy.GetParameterAsText(3)
+    if (EchelonField == "" or EchelonField is None):
+        EchelonField = "echelon"
+
+    # Used to infer the affiliation from the dataset name
+    dataPath = desc.path
+    datasetName = desc.name
+
+    # Get affiliation, needed because SIDC affiliation cannot always be derived from feature attributes
+    affiliation = arcpy.GetParameterAsText(4)
+    if (not (affiliation == "")) and (not affiliation is None) and (not affiliation in DictionaryConstants.validAffiliations) :
+        if (affiliation <> "NOT_SET") :
```

Comment:
Syntax error here with "<>": suggest "if (affiliation is not "NOT_SET") :" a little more pythonic too  :)

Reply:
At Python 3 (see note above about Python 2 only tools) it is a syntax error (no idea why I didn't use != ) not sure "is not" will work with constants. 

One note: for the toolbox/scripts I avoided making any changes to these to avoid having to rigorously retest - but I can make these changes if needed